src/models/unigram_model.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import tensorflow as tf
+
+from models.tf_model import TFModel
+from models.abstract_model import convert_set_to_input_and_target
+
+
+class UnigramModel(TFModel):
+    """Unigram model that uses word frequencies to compute word probabilities.
+
+    Use meta-training set to approximate word frequencies. During
+    evaluation, ignore each episode's support set and evaluate only on query set.
+    """
+
+    def __init__(self, config):
+        super(UnigramModel, self).__init__(config)
+
+    def _define_placedholders(self):
+        self._input_size = self._config['input_size']
+        self._time_steps = self._config['max_len']
+
+        self._words = tf.placeholder(
+            tf.int32, [None, self._time_steps - 1])
+        self._alpha = 1
+
+    def _build_graph(self):
+        word_count = tf.get_variable(
+            "word_count", [self._input_size],
+            initializer=tf.constant_initializer(self._alpha),
+            trainable=False)
+        flatten_words = tf.reshape(self._words, [-1])
+        ones = tf.zeros_like(flatten_words, dtype=tf.float32) + 1
+        self._train_op = tf.scatter_add(word_count, flatten_words, ones)
+
+        sum = tf.reduce_sum(word_count)
+        self._prob = tf.gather(word_count, flatten_words) / sum
+        self._avg_neg_log = -tf.reduce_mean(tf.log(self._prob))
+
+        self._prob_all = word_count / tf.reduce_sum(word_count)
```

Comment:
Any reason to not replace `tf.reduce_sum(word_count)` with `sum` here?

Reply:
No, good call.